hal/src/electron/modem/dns_client.cpp

```diff
@@ -433,7 +478,7 @@ int getHostByName(const char* name, MDM_IP* addr) {
             }
         } while (HAL_Timer_Get_Milli_Seconds() - timeStart < timeout);
         ++retries;
-    } while (retries <= MAX_RETRIES);
+    } while (retries <= MAX_DNS_QRY_RETRIES);
     return SYSTEM_ERROR_TIMEOUT;
```

Comment:
Given that the issue addressed in this PR is specific to LTE modems and we plan to use this DNS client on 2G/3G Electrons as well, I'd probably close the socket on unaffected platforms here.

Reply:
Since this is such an integral part of the system (DNS lookup for the Cloud and use of UDP/TCP Clients, how about we reserve this socket just like we do for the Cloud).   Sure that means one less socket for users, but there are still 5 additional sockets on Electrons (7 total) and that's quite a lot.

Reply:
Sure, that makes sense too :+1: Also, implementing this logic is outside the scope of this PR.